packages/mui-joy/src/Button/Button.tsx

```diff
@@ -1,21 +1,250 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_capitalize as capitalize, unstable_useForkRef as useForkRef } from '@mui/utils';
+import { useButton } from '@mui/base/ButtonUnstyled';
+import composeClasses from '@mui/base/composeClasses';
+import { styled, useThemeProps } from '../styles';
+import { rootShouldForwardProp } from '../styles/styled';
+import { ExtendButton, ButtonTypeMap, ButtonProps } from './ButtonProps';
+import buttonClasses, { getButtonUtilityClass } from './buttonClasses';
 
-export type ButtonProps = JSX.IntrinsicElements['button'];
+const useUtilityClasses = (ownerState: ButtonProps & { focusVisible: boolean }) => {
+  const { color, disabled, focusVisible, focusVisibleClassName, fullWidth, size, variant } =
+    ownerState;
 
-const Button = (props: ButtonProps) => {
-  return <button type="button" {...props} />;
+  const slots = {
+    root: [
+      'root',
+      disabled && 'disabled',
+      focusVisible && 'focusVisible',
+      fullWidth && 'fullWidth',
+      `variant${capitalize(variant!)}`,
+      `color${capitalize(color!)}`,
+      `size${capitalize(size!)}`,
+    ],
+  };
+
+  const composedClasses = composeClasses(slots, getButtonUtilityClass, {});
+
+  if (focusVisible && focusVisibleClassName) {
+    composedClasses.root += ` ${focusVisibleClassName}`;
+  }
+
+  return composedClasses;
 };
 
+const ButtonRoot = styled('button', {
+  name: 'JoyButton',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
```

Comment:
Not combining props in the classes makes sense, the purpose of doing it in v5 was to be compatible with v4.